### PR TITLE
Add option to enable debug messages

### PR DESCRIPTION
**Pull request by girlpunk**

### CODE REVIEW

**Comment by philippmeisberger** [comment]
The built-in python `logging` module could also be used for this.

File: src/files/pyfingerprint/pyfingerprint.py [hunk 1]
```diff
@@ -124,16 +125,19 @@ class PyFingerprint(object):
...
         Constructor
 
         @param string port
         @param integer baudRate
         @param integer(4 bytes) address
         @param integer(4 bytes) password
+        @param bool debug
         """
 
+        self.debug = debug
```
Comment (bastianraschke) [inline src/files/pyfingerprint/pyfingerprint.py:139]: why public?

File: src/files/pyfingerprint/pyfingerprint.py [hunk 2]
```diff
@@ -286,7 +290,8 @@ def __readPacket(self):
 
             if ( len(receivedFragment) != 0 ):
                 receivedFragment = self.__stringToByte(receivedFragment)
-                ## print 'Received packet fragment = ' + hex(receivedFragment)
+            if self.debug:
```
Comment (bastianraschke) [inline src/files/pyfingerprint/pyfingerprint.py:293]: Please use braces and explicit boolean expression (to be consistent with our code style)

File: src/files/pyfingerprint/pyfingerprint.py [hunk 3]
```diff
@@ -297,6 +302,8 @@ def __readPacket(self):
 
                 ## Check the packet header
                 if ( receivedPacketData[0] != self.__rightShift(FINGERPRINT_STARTCODE, 8) or receivedPacketData[1] != self.__rightShift(FINGERPRINT_STARTCODE, 0) ):
+                    if self.debug:
```
Comment (bastianraschke) [inline src/files/pyfingerprint/pyfingerprint.py:305]: Please use braces and explicit boolean expression (to be consistent with our code style)

File: src/files/pyfingerprint/pyfingerprint.py [hunk 4]
```diff
@@ -297,6 +302,8 @@ def __readPacket(self):
...
+                        print("Invalid packet recieved: "+str(receivedPacketData))
```
Comment (bastianraschke) [inline src/files/pyfingerprint/pyfingerprint.py:306]: typo + single marks please

File: src/files/pyfingerprint/pyfingerprint.py [hunk 5]
```diff
@@ -461,6 +468,7 @@ def setSystemParameter(self, parameterNumber, parameterValue):
         @param integer(1 byte) parameterValue
         @return boolean
         """
+        #TODO: Make seperate functions for each of the parameters, would be more useful.
```
Comment (bastianraschke) [inline src/files/pyfingerprint/pyfingerprint.py:471]: We implemented it according to the spec document that contained these call. It is the counterpart of the "getSystemParameters" call that gives us the full information without choice. Please remove TODO.

**Review by bastianraschke — CHANGES_REQUESTED** [review]
Yeah build in Python logging module would be quite better

**Comment by bastianraschke** [comment]
We closed the PR because of absence of @cyberjacob. The feature itself is nice, but without adjustments we don't want to merge it.

**Pull request closed without merging**